packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware.ts

```diff
@@ -99,34 +99,39 @@ const analyticsMiddleware = () => store => next => action => {
   const state = store.getState()
   const language = state.preferences.language
 
-  switch (action.type) {
-    case '@@INIT':
-      analytics.clear()
-      break
-    case AT.analytics.LOG_PAGE_VIEW:
-      analytics.push(AnalyticsKey.PAGE_VIEW, {
-        type: AnalyticsType.VIEW,
-        originalTimestamp: getOriginalTimestamp(),
-        referrer: referrer,
-        page: action.payload.route
-      })
-      break
-    case AT.modals.SHOW_MODAL:
-      analytics.push(AnalyticsKey.MODAL_VIEW, {
-        type: AnalyticsType.VIEW,
-        originalTimestamp: getOriginalTimestamp(),
-        referrer: location,
-        origin: modalOriginDictionary(action.payload.props.origin),
-        page: modalNameDictionary(action.payload.type)
-      })
-      break
-    case AT.components.identityVerification.INITIALIZE_VERIFICATION:
-      analytics.push(AnalyticsKey.KYC_INITIALIZE_VERIFICATION, {
-        type: AnalyticsType.EVENT,
-        originalTimestamp: getOriginalTimestamp(),
-        language: language
-      })
-      break
+  try {
+    switch (action.type) {
+      case '@@INIT':
+        analytics.clear()
+        break
+      case AT.analytics.LOG_PAGE_VIEW:
+        analytics.push(AnalyticsKey.PAGE_VIEW, {
+          type: AnalyticsType.VIEW,
+          originalTimestamp: getOriginalTimestamp(),
+          referrer: referrer,
+          page: action.payload.route
+        })
+        break
+      case AT.modals.SHOW_MODAL:
+        analytics.push(AnalyticsKey.MODAL_VIEW, {
+          type: AnalyticsType.VIEW,
+          originalTimestamp: getOriginalTimestamp(),
+          referrer: location,
+          origin: modalOriginDictionary(action.payload.props.origin),
+          page: modalNameDictionary(action.payload.type)
+        })
+        break
+      case AT.components.identityVerification.INITIALIZE_VERIFICATION:
+        analytics.push(AnalyticsKey.KYC_INITIALIZE_VERIFICATION, {
+          type: AnalyticsType.EVENT,
+          originalTimestamp: getOriginalTimestamp(),
+          language: language
+        })
+        break
+    }
+  } catch (e) {
+    // eslint-disable-next-line
```

Comment:
`analytics` is not `async`, so you can just not have that

Reply:
what do you mean?

Reply:
This line was catching and causing the app to stop responding to actions:

> modalOriginDictionary(action.payload.props.origin)

Reply:
This is actually causing multiple issues:

https://blockchain.atlassian.net/browse/FWLT-1339

Reply:
yeah i think this makes sense. gonna merge and deploy to unblock QA

Reply:
@pedroapfilho we can/may refactor all of this with Segment work

Reply:
AH, alright, my bad, I thought the `origin` was required